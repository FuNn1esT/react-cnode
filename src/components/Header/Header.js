import React from 'react';
import PropTypes from 'prop-types';
import { Layout, Menu, Dropdown, Avatar, Icon } from 'antd';
import { userUtils } from 'utils';
import styles from './Header.scss';

const menuConfig = [
  {
    key: 'all',
    name: '全部',
  },
  {
    key: 'good',
    name: '精华',
  },
  {
    key: 'share',
    name: '分享',
  },
  {
    key: 'ask',
    name: '问答',
  },
  {
    key: 'job',
    name: '招聘',
  },
  {
    key: 'dev',
    name: '测试',
  },
];

const Header = ({ onClickMenu }) => {
  const isLogin = userUtils.getUser().length > 0;
  let user;
  if (isLogin) {
    user = JSON.parse(userUtils.getUser());
  }
  const userMenu = (
    <Menu className={styles.dropDown} onClick={onClickMenu}>
      <Menu.Item key="logout"><Icon type="logout" />退出登录</Menu.Item>
    </Menu>
  );

  return (
    <Layout.Header className={styles.header}>
      <a href="/">
        <div className={styles.logo} />
      </a>
      <Menu
        className={styles.menu}
        mode="horizontal"
        defaultSelectedKeys={['all']}
        onClick={onClickMenu}
      >
        {menuConfig.map(item => (
          <Menu.Item key={item.key}>{item.name}</Menu.Item>
        ))}
      </Menu>
      {isLogin ?
        (
          <Dropdown overlay={userMenu}>
            <span className={styles.account}>
              <Avatar
                className={styles.avatar}
                size="small"
                src={user.avatar}
              />
              <span>{user.name}</span>
            </span>
          </Dropdown>
        ) :
        (
          <Menu
            className={styles.account}
            mode="horizontal"
            selectable={false}
            onClick={onClickMenu}
          >
            <Menu.Item key="login">登录</Menu.Item>
          </Menu>
        )}
    </Layout.Header>
  );
};
Header.propTypes = {
  onClickMenu: PropTypes.func.isRequired,
};

export default Header;
