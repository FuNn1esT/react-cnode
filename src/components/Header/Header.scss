@import "../../styles/Base";

.container {
  width: 100%;
  background: white;
  height: auto;
  display: flex;
  justify-content: center;
  box-shadow: 0 1px 4px rgba(0, 21, 41, .08);
}

.header {
  max-width: 100%;
  height: 100%;
  width: $screen-lg;
  display: flex;
  align-items: center;
  @media screen and(max-width: $screen-md) {
    flex-direction: column;
  }
}

.logo {
  width: 128px;
  height: 32px;
  float: left;
  margin: 16px 24px 16px 0;
  background: url("../../assets/logo.svg");
}

.menu {
  width: 100%;
  line-height: 64px;
  @media screen and(max-width: $screen-md) {
    display: flex;
    justify-content: center;
  }
}

.right {
  flex: 1;
  display: flex;
  justify-content: flex-end;
}

.account {
  line-height: 64px;
}

.avatar {
  margin-right: 8px;
}
