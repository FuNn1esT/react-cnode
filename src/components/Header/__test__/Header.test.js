import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Header from '../Header';

describe('<Header />', () => {
  it('should render login', () => {
    const props = {
      isMobile: false,
      user: {},
      onClickMenu: jest.fn(),
    };
    const tree = mount(
      <MemoryRouter>
        <Header {...props} />
      </MemoryRouter>,
    );
    expect(toJson(tree, { noKey: true, mode: 'deep' })).toMatchSnapshot();
  });
});
