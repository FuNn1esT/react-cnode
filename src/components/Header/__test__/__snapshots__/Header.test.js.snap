// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render login 1`] = `
<div
  className="ant-layout-header"
>
  <div
    className=""
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div />
    </a>
    <ul
      className="ant-menu ant-menu-light ant-menu-root ant-menu-horizontal"
      onClick={[Function]}
      role="menu"
      style={Object {}}
    >
      <li
        className="ant-menu-item ant-menu-item-selected"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={Object {}}
      >
        全部
      </li>
      <li
        className="ant-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={Object {}}
      >
        精华
      </li>
      <li
        className="ant-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={Object {}}
      >
        分享
      </li>
      <li
        className="ant-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={Object {}}
      >
        问答
      </li>
      <li
        className="ant-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={Object {}}
      >
        招聘
      </li>
    </ul>
    <div>
      <ul
        className="ant-menu ant-menu-light ant-menu-root ant-menu-horizontal"
        onClick={[Function]}
        role="menu"
        style={Object {}}
      >
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          登录
        </li>
      </ul>
    </div>
  </div>
</div>
`;
